docs/guides.rst

```diff
@@ -304,3 +304,77 @@ decorator.
 
 When a method decorated with :func:`~async_service.external_asyncio_api` fails
 it raises an :class:`async_service.exceptions.ServiceCancelled` exception.
+
+Cleanup logic
+-------------
+
+In the case that we need to run some logic **after** the service has finished
+running but **before** the service has registered as finished we can do so with
+the following patterns.  However, special care and consideration should be
+taken as the following patterns can result in the application hanging when we
+try to shut it down.
+
+The basic idea is to use a ``try/finally`` expression in we main
```

Comment:
s/we/our/

Reply:
That's the right sized PR to come back from holidays :sweat_smile: 
https://github.com/ethereum/async-service/pull/42